include/snap/metric.h

```diff
@@ -15,32 +15,68 @@ limitations under the License.
 #ifndef METRIC_H
 #define METRIC_H
 
-#include <ctime>
+#include <chrono>
 #include <map>
 #include <string>
 #include <vector>
 
+#include <google/protobuf/repeated_field.h>
+
+#include "plugin.pb.h"
+
+#include "snap/config.h"
+
+using namespace ::google::protobuf;
+
 namespace Metric {
 
-class Namespace : private std::vector<std::string> {
+struct NamespaceElement {
+  public:
+    std::string value;
+    std::string name;
+    std::string description;
 };
 
-// TODO (danielscottt): use a union for data here.
-class Metric {
-
+class Metric final {
   public:
-    void setData(int data);
+    Metric(std::vector<NamespaceElement> ns, std::string unit, std::string descrip);
```

Comment:
I think `descrip` should probably be either `desc` or `description`, leaning heavily towards the latter. Can put on multiple lines if the width is to long.